src/modules/LR11x0/LR11x0.cpp

```diff
@@ -1282,14 +1282,13 @@ uint32_t LR11x0::getTimeOnAir(size_t len) {
     }
 
     // get time-on-air in us
-    return(((uint32_t(1) << this->spreadingFactor) / this->bandwidthKhz) * N_symbol * 1000.0);
+    return(((RadioLibTime_t(1) << this->spreadingFactor) / this->bandwidthKhz) * N_symbol * 1000.0);
 
   } else if(type == RADIOLIB_LR11X0_PACKET_TYPE_GFSK) {
-    return(((uint32_t)len * 8 * 1000000UL) / this->bitRate);
+    return(((RadioLibTime_t)len * 8 * 1000000UL) / this->bitRate);
   
   } else if(type == RADIOLIB_LR11X0_PACKET_TYPE_LR_FHSS) {
-    return(((uint32_t)len * 8 * 1000000UL) / RADIOLIB_LR11X0_LR_FHSS_BIT_RATE);
-
+    return(((RadioLibTime_t)len * 8 * 1000000UL) / RADIOLIB_LR11X0_LR_FHSS_BIT_RATE);
```

Comment:
I don't think these casts should be to `RadioLibTime_t`, the point here is to not overflow `int` with the bitshifts and multiplications. HAL millis()/micros() are not used here, so `uint32_t` should stay internally in this method. The return type is correct, as outside it needs to be trated as time.

This is also the case for `SX127x::getTimeOnAir` and `SX128x::getTimeOnAir`.

Reply:
Done in e35e03